### DISCUSSION TITLE
Unable to create multiple calendar

### DISCUSSION THREAD

**Discussion by monsieurlee** [derekantrican/GAS-ICS-Sync#334] [Q&A] [answered]
Hi, not sure if this is a bug, so I'd post this here first.

We have 9 .ics feed that I'm trying to set up the script for. I set up the script one using a single .ics, and everything ran great (thank you for creating this script and sharing it!). A calendar is created in GCal, and all the events show up and gets updated.

After I got the first working, I added 8 more feed. However, when I run the script, Only 1 other calendar gets added to GCal. The rest doesn't get added. After some trial and errors, I go to GCal to manually add the calendar, and the site won't add me add more than 2 until I reload, after which I can manually add the rest. So, this makes me think the problem is on Google's end.

I named the manually added calendars the same name as the ones in the script, but they don't sync. which did not surprise me. Maybe the mapping is based on more than the name.

I looked at the Calendars individually. The one created by the script says "Created by GAS" and has a calendar ID this long:
c_3or57pirg1n3ldeaah6a4i@group.calendar.google.com
All the rest I created manually have a calendar ID this long:
c_7b440f63005742b377509da63f2cd38151e4267ad4894c84a5a921f58d3b@group.calendar.google.com

So I'm wondering:
- Does this issue sound like a bug?
- Is there a limit on the number of calendars I can use? I assumed at first because line 28 of the code showed 3 .ics feed examples, and after I ran into this issue, I searched the issues and found [#62 ](https://github.com/derekantrican/GAS-ICS-Sync/issues/62) linking to #63 showing that multiple calendar is supported. Is there a hard limit?
- If the script is unable to create 9 calendars initially maybe due to some error, is there a way to map the ones that I created manually (with a longer ID) with the .ics feed in the script, if they are connected by more than just the name?

Thank you!

**Comment by jonas0b1011001** [comment]
> * Does this issue sound like a bug?

Definitely! We just need to find out if it's on our side or on google's :)



> * Is there a limit on the number of calendars I can use?

No.



> * If the script is unable to create 9 calendars initially maybe due to some error, is there a way to map the ones that I created manually (with a longer ID) with the .ics feed in the script, if they are connected by more than just the name?

We are actually just matching names..
Syncing to manually created calendars is also supported, the script checks and will select any calendar your account has write access to.

> and the site won't add me add more than 2 until I reload, after which I can manually add the rest.

I have experienced that before, seems to be a temporary issue though.

---
---
Do you by chance have more than 250 calendars in your account?

**Reply by monsieurlee** [reply]
> We are actually just matching names.. Syncing to manually created calendars is also supported, the script checks and will select any calendar your account has write access to.

Hmm, when I tried to sync to manually created calendars, it wasn't syncing.

I don't know if this matters, but the manually created calendars all have only all day events, whereas the automatically created calendar that worked has "normal" events that last a few hours. Would that made a difference?
 
> > and the site won't add me add more than 2 until I reload, after which I can manually add the rest.
> 
> I have experienced that before, seems to be a temporary issue though.
> 
> Do you by chance have more than 250 calendars in your account?

No, I only have 2 calendars before I ran the script.

So I removed the trigger, and removed all of the manual and 1 auto-added calendars from the script so I can start over. Once again, after I ran install, and nothing happened for a few minutes, then I ran startSync, and the 2 (out of 9) calendars are added. the first one worked fine with all the hourly events showed up. the 2nd one, with the all day events, is blank. Just to confirm when I add it as URL under Other Calendar, then it shows up

**Reply by jonas0b1011001** [reply]
> Would that made a difference?

That should not matter. Are all of your urls working if you sync them individually?




> then I ran startSync

Can you share the log of that execution? You can view them at [https://script.google.com/home/executions](https://script.google.com/home/executions)

**Reply by monsieurlee** [reply]
> > Would that made a difference?
> 
> That should not matter. Are all of your urls working if you sync them individually?

I will try this next


> > then I ran startSync
> 
> Can you share the log of that execution? You can view them at https://script.google.com/home/executions

Looks like it is throwing an error on the 2nd calender, the one named "ASBH Desk 1a".
This is the feed. I ran it through the iCalendar Validator and I didn't get any errors
https://anchorspacebarharbor.com/ics/desk-1a/wpbm.ics

![Screenshot 2023-04-13 at 15-34-32 GAS-ICS-Sync ASBH - Project Executions - Apps Script](https://user-images.githubusercontent.com/1504053/231865703-ade2f0f3-c507-4f33-b3c0-2919d469803c.png)


Thank you

**Comment by jonas0b1011001** [comment, marked as answer]
This turns out to be the same issue as #258.
The start of the event is defined as date & time, the end is date only.

Please do the following changes in **helpers.gs** and try again to sync.
* Change line [280](https://github.com/derekantrican/GAS-ICS-Sync/blob/e5cd8e02417ee38a04eadbd795201822dbac2e2d/Helpers.gs#L280) to
`if(icalEvent.startDate.isDate && icalEvent.endDate.isDate){ //All-day event`
* add to line [306](https://github.com/derekantrican/GAS-ICS-Sync/blob/e5cd8e02417ee38a04eadbd795201822dbac2e2d/Helpers.gs#L306)
```
icalEvent.startDate.isDate = false;
icalEvent.endDate.isDate = false;
```

**Comment by monsieurlee** [comment]
Thank worked. Thank you so much!

I'm keep an eye on it for the next little while and do some testing on updating events and what not.

Really appreciate the help on this.
